src/runtime_src/core/edge/user/shim.cpp

```diff
@@ -153,7 +153,18 @@ int ZYNQShim::mapKernelControl(const std::vector<std::pair<uint64_t, size_t>>& o
     if ((offset_it->first & (~0xFF)) != (-1UL & ~0xFF)) {
       auto it = mKernelControl.find(offset_it->first);
       if (it == mKernelControl.end()) {
-        ptr = mmap(0, offset_it->second, PROT_READ | PROT_WRITE, MAP_SHARED, mKernelFD, offset_it->first);
+        drm_zocl_info_cu info = {offset_it->first, -1};
+        int result = ioctl(mKernelFD, DRM_IOCTL_ZOCL_INFO_CU, &info);
+        if(result) {
+            printf("failed to find CU info 0x%lx\n", offset_it->first);
+            return -1;
+        }
+        size_t psize = getpagesize();
+        if(info.apt_idx == -EINVAL) {
+            printf("Failed to find CU in aperture list 0x%lx\n", offset_it->first);
+            return -1;
+        }
```

Comment:
As commented, this seems not necessary. just fail the ioctl if we can not find the CU physical address in kernel

Reply:
removed